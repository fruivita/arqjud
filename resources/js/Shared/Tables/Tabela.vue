<!--
    Tabela (table) padrão.

    Notar que o componente depende do status do processamento geral da
    aplicação para definir seu próprio estado.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import { useStatusRequisicaoStore } from '@/Stores/StatusRequisicaoStore';

const status = useStatusRequisicaoStore();
</script>

<template>
    <div class="overflow-x-auto">
        <table
            :class="{
                'opacity-50': status.processando === true,
            }"
            class="w-full text-center"
        >
            <thead class="bg-primaria-200 dark:bg-secundaria-700">
                <tr dusk="header">
                    <slot name="header"></slot>
                </tr>
            </thead>

            <tbody dusk="body">
                <slot name="body"></slot>
            </tbody>
        </table>
    </div>
</template>
