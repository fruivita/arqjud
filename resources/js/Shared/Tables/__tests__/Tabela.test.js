/**
 * Testes para o componente Tabela.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Tabela from '@/Shared/Tables/Tabela.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;
let pinia;

beforeEach(() => {
    pinia = (options = {}) =>
        createTestingPinia({
            initialState: {
                StatusRequisicaoStore: { ...options },
            },
        });

    mountFunction = (options = {}) => {
        return mount(Tabela, { ...options });
    };
});

describe('Tabela', () => {
    // Caminho feliz
    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { header: 'foo', body: 'bar' },
                global: {
                    plugins: [pinia({ processando: false })],
                },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente de maneira diversa enquanto a pÃ¡gina se carrega respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { header: 'foo', body: 'bar' },
                global: {
                    plugins: [pinia({ processando: true })],
                },
            }).html()
        ).toMatchSnapshot();
    });
});
