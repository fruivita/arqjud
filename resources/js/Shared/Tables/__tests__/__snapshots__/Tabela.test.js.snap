// Vitest Snapshot v1

exports[`Tabela > renderiza o componente de maneira diversa enquanto a pÃ¡gina se carrega respeitando o snapshot 1`] = `
"<div class=\\"overflow-x-auto\\">
  <table class=\\"opacity-50 w-full text-center\\">
    <thead class=\\"bg-primaria-200 dark:bg-secundaria-700\\">
      <tr dusk=\\"header\\">foo</tr>
    </thead>
    <tbody dusk=\\"body\\">bar</tbody>
  </table>
</div>"
`;

exports[`Tabela > renderiza o componente respeitando o snapshot 1`] = `
"<div class=\\"overflow-x-auto\\">
  <table class=\\"w-full text-center\\">
    <thead class=\\"bg-primaria-200 dark:bg-secundaria-700\\">
      <tr dusk=\\"header\\">foo</tr>
    </thead>
    <tbody dusk=\\"body\\">bar</tbody>
  </table>
</div>"
`;
